### PR TITLE
feat: toIndentYAML -support indentation for YAML output

### PR DESCRIPTION
**Pull request by mrueg**
## Description
Resolves issue described in https://github.com/go-sprout/sprout/issues/79 where a yaml function that supports Indentation is needed.

## Changes
* Add new function toIndentYaml

## Fixes 
Fixes https://github.com/go-sprout/sprout/issues/79
## Checklist
- [x] I have read the **CONTRIBUTING.md** document.
- [x] My code follows the code style of this project.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have updated the documentation accordingly.
- [x] This change requires a change to the documentation on the website.

### CODE REVIEW

File: docs/registries/encoding.md
Comment (mrueg) [inline docs/registries/encoding.md:183]: This needs a better example here, I currently don't know how nested dicts can be set up easily (need to look it up)

File: registry/encoding/functions.go
Comment (mrueg) [inline registry/encoding/functions.go:252]: Update example here as well.

File: registry/encoding/functions.go
Comment (42atomys) [inline registry/encoding/functions.go:272]: We can reduce memory usage and avoid casting by passing the result directly
```suggestion
	return strings.TrimSuffix(buf.String(), "\n"), nil
```

File: registry/encoding/functions.go
Comment (42atomys) [inline registry/encoding/functions.go:263]: We can use the one line condition here to have the same style everywhere
```suggestion
	if err = enc.Encode(&v); err != nil {
```

**Review by 42atomys — APPROVED** [review]
Two small suggestions, but it looks good to me in general. ✅

I approve it and will keep it on hold for documentation after merging #86!

Thanks for your contribution. 💜

File: registry/encoding/functions_test.go
Comment (ccoVeille) [inline registry/encoding/functions_test.go:113]:
```suggestion
		{Name: "TestEmptyInput", Input: `{{ "" | toIndentYaml 8 }}`, ExpectedOutput: `""`},
```

File: registry/encoding/functions.go
Comment (ccoVeille) [inline registry/encoding/functions.go:267]:
```suggestion
 		return "", fmt.Errorf("YAML encode error: %w", err)
```

I know Go error are usually in lowercase, but YAML is an acronym

Comment (42atomys) [reply registry/encoding/functions.go:267]: Go error should start with a lowercase, so that will me strange to do `yAML`, This is a really good question.. I will search about it and come back with final answer !

Comment (ccoVeille) [reply registry/encoding/functions.go:267]: Not exactly

https://go.dev/wiki/CodeReviewComments#error-strings

> Error strings should not be capitalized (unless beginning with proper nouns or acronyms) or end with punctuation, since they are usually printed following other context. That is, use fmt.Errorf("something bad") not

Comment (42atomys) [reply registry/encoding/functions.go:267]: Too fast for me 👏 Agree with `YAML`

File: registry/encoding/functions_test.go
Comment (mrueg) [reply registry/encoding/functions_test.go]: Made the change, thanks!

**Pull request merged**
